client/src/test/java/com/paypal/selion/reports/reporter/excelreport/ExcelReportTest.java

```diff
@@ -59,6 +84,34 @@ public void testExcelReporter() {
         iSuites.add(iSuite);
 
         ExcelReport excelReport = new ExcelReport();
+        excelReport.setExcelFileName(excelReportFileName);
         excelReport.generateReport(xmlSuites, iSuites, SUREFIRE_DEFAULT_REPORTS_DIR);
+
+        // Check whether the ExcelReport exists.
+        assertTrue(Files.exists(p), "Verify path of Excel Report file.");
+        assertTrue(Files.isRegularFile(p), "Verify that Excel_Report.xls is a file.");
+
+        if (Files.exists(p) && Files.isRegularFile(p)) {
+            FileInputStream fileInputStream = null;
+            HSSFWorkbook workbook = null;
+            try {
+                fileInputStream = new FileInputStream(p.toFile());
+                workbook = new HSSFWorkbook(fileInputStream);
+                assertEquals(ReportSheetNames.values().length, workbook.getNumberOfSheets(),
+                        "Verify number of worksheets in ExcelReport");
+            } catch (FileNotFoundException e) {
```

Comment:
I dont think we resort to this practice of catching exceptions and explicitly failing the tests. By default TestNG fails a test if it throws an exception. So we could perhaps clear up this part.